app/models/line_item.rb

```diff
@@ -0,0 +1,26 @@
+class LineItem < ApplicationRecord
+  has_many :orders_line_items
+  has_many :orders, through: :orders_line_items
+  belongs_to :cart
+
+  validates :code, :quantity, :price, :total, :cart_id, presence: true
+  validates :code, uniqueness: true
+
+  def update_total(pricing_rules)
+    self.total = pricing_rules.apply(self) || add_item_price_to_total
+  end
+
+  def price
+    read_attribute(:price).to_f if read_attribute(:price)
+  end
+
+  def total
+    read_attribute(:total).to_f if read_attribute(:total)
+  end
```

Comment:
Here I ensured that future authors of this code could theoretically read its results more easily when either relying on it for more backend features, debugging, or testing.

Once consumed via an api or otherwise this will be more human readable even without this, but as mentioned this for purposes of use within the backend.